src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,80 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useDispatch } from 'react-redux';
+import { setCurrentTodo } from '../../features/currentTodo';
+import { useAppSelector } from '../../app/hooks';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+export const TodoModal = () => {
+  const dispatch = useDispatch();
+  const todo = useAppSelector(state => state.currentTodo);
 
-      <Loader />
+  const { id, title, completed, userId } = todo || {};
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+  useEffect(() => {
+    if (userId) {
+      setIsLoading(true);
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+      getUser(userId)
+        .then(setUser)
+        .catch(error => {
```

Comment:
You are not handling the error from the `getUser` function properly. Throwing an error inside a promise `.catch()` handler will not stop the execution of the program, but will simply generate an unhandled promise rejection. Instead, consider displaying an error message to the user or implementing a retry mechanism.